### ISSUE TITLE
Immediate Jobs are never detected as stalled or broken

### ISSUE THREAD

**Issue by andrewandante** [silverstripe-australia/silverstripe-queuedjobs#355]
Jobs in the immediate queue will stick in "Running" forever (for whatever reason, haven't narrowed that one down yet) but then if they do, they are never detected as "stalled". I assume this is due to the different way the Immediate queue is handled?

**Comment by emteknetnz** [comment]
I've seen this behaviour, it's pretty annoying

I don't know the reasons why, though it would be nice if it said stalled or broken instead

**Comment by andrewandante** [comment]
Yeah I've since found that it's only a problem when using the "on shutdown" version of the Immediate queue - I think it needs to still go through some semi-queue actions when this happens, otherwise you get this.
